src/odemis/driver/smarpod.py

```diff
@@ -279,9 +279,256 @@ def __init__(self, error_code):
+
+    # handles
+    # handle value that means no object
+    SA_MC_INVALID_HANDLE = 0xffffffff
+    SA_MC_INFINITE = -1
+
+    err_code = {
+0x0000: "No error",
+0x0001: "Unspecified error",
+0x0002: "Invalid parameter in function call ",
+0x0003: "Invalid locator in call to Open function ",
+0x0004: "Undefined or inaccessible property in function call ",
+0x0005: "Invalid handle in call to function ",
+0x0006: "Tried to use an unspported feature",
+0x0007: "Reached limit of simultaneously controllable devices ",
+0x0008: "Supplied buffer too small",
+0x0100: "An operation has been canceled while waiting for a result ",
+0x0101: "An operation has timed out ",
+0x0200: "The pose specified in the Move command is invalid/unreachable ",
+0x0201: "Device has not been referenced ",
+0x0203: "An operation could not be started because the device is busy ",
+0x0300: "Positioners were blocked during movement ",
+0x0301: "The following error limit has been exceeded during movement ",
+0x0320: "Positioner referencing failed ",
+0x0500: "Could not load required hardware driver",
+0x0501: "Could not find/connect to controller",
+0x0501: "Could not find/connect to controller",
+0x0502: "The device is not connected",
+0x0503: "The controller doesn't provide the require features or configuration",
+0x0504: "Error when communicating with controller",
+        }
+
+    def __init__(self):
+        if os.name == "nt":
+            raise NotImplemented("Windows not yet supported")
+            # WinDLL.__init__(self, "libSA_MC.dll")  # TODO check it works
+            # atmcd64d.dll on 64 bits
+        else:
+            # Global so that its sub-libraries can access it
+            CDLL.__init__(self, "libsmaractmc.so.0.9.1", RTLD_GLOBAL)
+
+    def __getitem__(self, name):
+        try:
+            func = super(SA_MCDLL, self).__getitem__(name)
+        except Exception:
+            raise AttributeError("Failed to find %s" % (name,))
+        func.__name__ = name
+        func.errcheck = self.sp_errcheck
+        return func
+
+    @staticmethod
+    def sp_errcheck(result, func, args):
+        """
+        Analyse the retuhwModelrn value of a call and raise an exception in case of
+        error.
+        Follows the ctypes.errcheck callback convention
+        """
+        # everything returns DRV_SUCCESS on correct usage, _except_ GetTemperature()
+        if result != SA_MCDLL.SA_MC_OK:
+            raise SA_MCError(result)
+
+        return result
+
+
+class SA_MCError(Exception):
+    """
+    SA_MC Exception
+    """
+
+    def __init__(self, error_code):
+        self.errno = error_code
+        super(SA_MCError, self).__init__("Error %d. %s" % (error_code, SA_MCDLL.err_code.get(error_code, "")))
+
+
 class SmarPod(model.Actuator):
     
-    def __init__(self, name, role, locator, ref_on_init=False, actuator_speed=0.1,
+    def __init__(self, name, role, locator, daemon=None, ref_on_init=False, actuator_speed=0.1,
```

Comment:
Don't get `daemon` here. It is normally a hidden data directly passed to the parent class (via kwargs). If you grab it here, the parent will never get it and be very sad.